### PR TITLE
Cleanup/drop monix types pkg

### LINKED ISSUE #5: monix-types sub-project serves no purpose
The `funfix-types` sub-project was exposing type classes, similar to [Cats](https://github.com/typelevel/cats), then moved in https://funland-js.org.

I do not see a use-case for having a `monix-types` for now.

It's a little confusing, plus in its current form it actually limits the modularity of the project. If we are to add for example a `monix-tail` sub-project, exposing an `Iterant`, like in Monix, then the `Observer` data type is kind of useless for `monix-tail`. `Ack` as well.

So sub-projects have to be either reusable between Monix's sub-projects, or be useful on their own. I don't see this for `monix-type` at this time.

### LINKED ISSUE #6: WithScheduler interface
Currently you're describing `Observer` like this:

```typescript
export interface Observer<T> {
  onNext(elem: T): Ack

  onComplete(): void

  onError(e: Throwable): void
}

export interface SyncObserver<T> extends Observer<T> {
  onNext(elem: T): SyncAck
}

export interface WithScheduler {
  readonly scheduler: Scheduler
}

export interface Subscriber<T> extends WithScheduler, Observer<T> {
}

export interface SyncSubscriber<T> extends WithScheduler, SyncObserver<T> {
}
```

Given that TypeScript is structurally typed, the `WithScheduler` serves no purpose and should be removed.

There's never a case for a function to require only a `WithScheduler` object and even if it were, you can always come up with it on the fly.

We should be wary of the symbols we export and only export those that are absolutely necessary.

### PR DESCRIPTION
**Pull request by lostintime**
Drops `monix-types` package (resolves #5)
Drops `WithScheduler` interface (resolves #6)
Drops `IObservable` interface

`Observable` now extends `ObservableInstance` to hide later for clients, ex: 
```typescript
import { Observable } from "monix"

function doSomethingOn<A>(a: Observable<A>): Observable<A> {
  return a
}
```

Is it worth it to create flow definitions for internals, while those not exposed to clients?

What is the right way to run `test:watch` using `lerna`, `lerna run --scope=monix-reactive test:watch` doesn't seem to work?

How to get rid of `flow-typed` folders? not it fails with:

```
Cannot resolve module `funfix`.

   20| import type { Future } from "funfix"
                                   ^^^^^^^^
```

`node_modules` removed from ignored in [.flowconfig](https://github.com/funfix/monix.js/blob/cleanup/drop_monix_types_pkg/common/.flowconfig)

### CODE REVIEW

**Comment by lostintime** [comment]
Added `.travis.yml` to run build, test and upload code coverage (resolves #9)

**Comment by alexandru** [comment]
Your PRs are too big. Big PRs are very difficult to review.

Given you have energy and bandwidth to work on this, might I suggest a workflow like this ...

1) first you create a branch

```
git checkout -b feature1
```

You work on it, then you create a PR.
And while that PR is pending review ...

2) you fork another branch from `feature1`

```
git checkout feature1

git checkout -b feature2
```

You work on it, but wait on the PR for `feature1`, then you can create the PR for `feature2`.

3) you for another branch from `feature2`

Etc, etc...

**Comment by alexandru** [comment]
Now that you've touched on this, I'm not very happy about `ObservableInstance` versus `Observable`.

From what I understand `Observable` has static methods in it. Maybe those static methods should not be described on the `Observable` class.

Maybe we need an `Observables` object (not class).

Here's the point at which we need to investigate what other libraries are doing:

1. latest `master` on https://github.com/Reactive-Extensions/RxJS
2. https://github.com/cujojs/most

I see for example that `most` works with plain functions. Maybe that's not bad at all.

**Comment by lostintime** [comment]
Sure, it will settle, those are bigger changes was hard to split, like dropping `IObservable`. Also I have a schedule for this work and today - is the day :).
Will try to split upcoming changes to smaller, atomic requests.

**Comment by alexandru** [comment]
Btw, to talk about this issue, I opened another issue — https://github.com/funfix/monix.js/issues/10

Finish what you were doing in this PR first, we can merge it as is to finish the removal of `monix-types` sub-project.

Also — add a "*WIP:*" for as long as you work on a PR and remove it when it's ready to be merged.

**Comment by alexandru** [comment]
@lostintime OK, cool. So yeah, I don't want to stop you from working on this 😀

Was just mentioning that PRs work when they come in a smaller size, otherwise it gets hard to review.

If you want to beat this into some shape, I won't stop you for now, we can talk about changes later.

It's just that I don't want you to then get discouraged if we talk about undoing what you've worked on already.

So go ahead while you still have time and we'll figure it out.

**Comment by lostintime** [comment]
@alexandru no problem with undoing, while it makes it better :).
For now it will be great to iterate over core things like https://github.com/funfix/monix.js/issues/10 so the next PRs may be done one per builder/operator, not touching lots of files.

This PR is removing `monix-types` package and not going to push more changes here. It also includes CI configuration and drops `IObservable` which was wrong move, next time will use separate PRs.

As next steps I'm going to check https://github.com/funfix/monix.js/issues/10, remove some useless flow files and improve code coverage.

File: packages/monix-reactive/package.json [hunk 1]
```diff
@@ -61,6 +61,6 @@
     "reactive streams"
   ],
   "dependencies": {
-    "monix-types": "0.0.0-development"
+    "funfix": "^6.2.2"
```
Comment (alexandru) [inline packages/monix-reactive/package.json]: I should really publish version 7 already, it's at release candidate. Will let you know.

**Pull request merged**